cmd/common/trace/signal.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	commonutils "github.com/kinvolk/inspektor-gadget/cmd/common/utils"
+	signalTypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/signal/types"
+	"github.com/spf13/cobra"
+)
+
+type SignalFlags struct {
+	Pid    uint64
+	Sig    string
+	Failed bool
+}
+
+func newSignalParser(outputConfig *commonutils.OutputConfig, options ...commonutils.Option) TraceParser[signalTypes.Event] {
+	return commonutils.NewGadgetParser(outputConfig, signalTypes.MustGetColumns(), options...)
+}
+
+func NewSignalParserWithK8sInfo(outputConfig *commonutils.OutputConfig) TraceParser[signalTypes.Event] {
+	return newSignalParser(outputConfig, commonutils.WithMetadataTag(commonutils.KubernetesTag))
+}
+
+func NewSignalParserWithRuntimeInfo(outputConfig *commonutils.OutputConfig) TraceParser[signalTypes.Event] {
+	return newSignalParser(outputConfig, commonutils.WithMetadataTag(commonutils.ContainerRuntimeTag))
+}
+
+func NewSignalParser(outputConfig *commonutils.OutputConfig) TraceParser[signalTypes.Event] {
+	return newSignalParser(outputConfig)
+}
```

Comment:
How about using generic functions instead? Avoids having dedicated functions for each gadget by just passing one more argument instead.

```golang
func NewParser[T TraceEvent](outputConfig *commonutils.OutputConfig, columns *columns.Columns[T]) TraceParser[T] {
	return commonutils.NewGadgetParser[T](outputConfig, columns)
}

func NewParserWithK8sInfo[T TraceEvent](outputConfig *commonutils.OutputConfig, columns *columns.Columns[T]) TraceParser[T] {
	return commonutils.NewGadgetParser[T](outputConfig, columns, commonutils.WithMetadataTag(commonutils.KubernetesTag))
}

func NewParserWithRuntimeInfo[T TraceEvent](outputConfig *commonutils.OutputConfig, columns *columns.Columns[T]) TraceParser[T] {
	return commonutils.NewGadgetParser[T](outputConfig, columns, commonutils.WithMetadataTag(commonutils.ContainerRuntimeTag))
}
```

Reply:
Even though I was trying to hide the call to `types.GetColumns`, having these four functions for each gadget creates a lot of duplicated code and it is not worth it.